### PR TITLE
web/satellite common search component created

### PR DESCRIPTION
**Pull request by VitaliiShpital**
What: web/satellite common search component created

Why:

Please describe the tests:
 - Test 1:
 - Test 2:
 
Please describe the performance impact:

## Code Review Checklist (to be filled out by reviewer)
 - [ ] Does the PR describe what changes are being made?
 - [ ] Does the PR describe why the changes are being made?
 - [ ] Does the code follow [our style guide](https://github.com/storj/docs/blob/master/code/Style.md)?
 - [ ] Does the code follow [our testing guide](https://github.com/storj/docs/blob/master/code/Testing.md)?
 - [ ] Is the PR appropriately sized? (If it could be broken into smaller PRs it should be)
 - [ ] Does the new code have enough tests? (*every* PR should have tests or justification otherwise. Bug-fix PRs especially)
 - [ ] Does the new code have enough documentation that answers "how do I use it?" and "what does it do?"? (both source documentation and [higher level](https://github.com/storj/docs), diagrams?)
 - [ ] Does any documentation need updating?
 - [ ] Do the database access patterns make sense?

### CODE REVIEW

File: web/satellite/src/components/common/SearchComponent.vue [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+</template>
+
+<script lang="ts">
+    import { Component, Prop, Vue } from 'vue-property-decorator';
+
+    declare type searchCallback = (search: string) => Promise<void>;
+
+    @Component
+    export default class SearchComponent extends Vue {
+        @Prop({default: ''})
```
Comment (VinozzZ) [inline web/satellite/src/components/common/SearchComponent.vue]: Is line 24 not enough for declaring the `default` prop?

File: web/satellite/tests/unit/common/__snapshots__/SearchComponent.spec.ts.snap [hunk 2]
```diff
@@ -0,0 +1,3 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
```
Comment (VinozzZ) [inline web/satellite/tests/unit/common/__snapshots__/SearchComponent.spec.ts.snap]: Is this file expected to be in here?

File: web/satellite/src/components/common/SearchComponent.vue (hunk 1 above)
Comment (VitaliiShpital) [reply web/satellite/src/components/common/SearchComponent.vue]: i'm not sure if i understand you right

there is '@Prop' decorator for each prop we have. Each prop has it's own default value so default value should be declared in every '@Prop' decorator.

here 24 line default value stands for 25 line prop and 22 line default value stands for 23 line prop.

Did you mean something else?

File: web/satellite/tests/unit/common/__snapshots__/SearchComponent.spec.ts.snap (hunk 2 above)
Comment (VitaliiShpital) [reply web/satellite/tests/unit/common/__snapshots__/SearchComponent.spec.ts.snap]: this file was generated automatically during testing. Should i remove this line?

File: web/satellite/src/components/common/SearchComponent.vue (hunk 1 above)
Comment (VinozzZ) [reply web/satellite/src/components/common/SearchComponent.vue]: A-ha, I see. You got it just right! Thanks for clarifying for me

File: web/satellite/tests/unit/common/__snapshots__/SearchComponent.spec.ts.snap (hunk 2 above)
Comment (VinozzZ) [reply web/satellite/tests/unit/common/__snapshots__/SearchComponent.spec.ts.snap]: IMO, I don't think we need to have source control for snapshots, but maybe we can see what others think?

Comment (crawter) [reply web/satellite/tests/unit/common/__snapshots__/SearchComponent.spec.ts.snap]: We need this snapshots to be able to compare that our markup has been changed.

The Jest [docs](https://jestjs.io/docs/en/snapshot-testing.html) also say that "the snapshot artifact should be committed alongside code changes"

Also, this means that the tests won't have a baseline for comparison when run in a fresh environment or during CI

Comment (VinozzZ) [reply web/satellite/tests/unit/common/__snapshots__/SearchComponent.spec.ts.snap]: Oh, I didn't know we are going to set up running the tests on CI now. That makes sense!

Comment (crawter) [reply web/satellite/tests/unit/common/__snapshots__/SearchComponent.spec.ts.snap]: @VinozzZ 
Yeah, we are going to make some frontend improvements at the beginnig of september. Adding linter/unit test/auto tests (yes, we have them :D ) to our CI also part of it

File: web/satellite/tests/unit/common/SearchComponent.spec.ts [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+        expect(wrapper.find('input').props('placeHolder')).toMatch('');
+        expect(wrapper.find('input').props('search')).toMatch('');
+    });
+
+    it('functions onMouseEnter/onMouseLeave work correctly', () => {
+        const wrapper = mount(SearchComponent);
+
+        wrapper.vm.onMouseEnter();
+
+        expect(wrapper.vm.inputWidth).toMatch('602px');
```
Comment (VinozzZ) [inline web/satellite/tests/unit/common/SearchComponent.spec.ts]: Seems like this test broke the build @VitaliiShpital . If we want to access private field, we need to use `vm.['inputWidth']`

Comment (crawter) [reply web/satellite/tests/unit/common/SearchComponent.spec.ts]: Do u mean that npm run build command fails?
its strange, because npm run test passes

**Pull request merged**